clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -480,18 +480,6 @@ private static String parseAcks(String acksString) {
         return newConfigs;
     }
 
-    public static Properties addSerializerToConfig(Properties properties,
-                                                   Serializer<?> keySerializer,
-                                                   Serializer<?> valueSerializer) {
-        Properties newProperties = new Properties();
-        newProperties.putAll(properties);
-        if (keySerializer != null)
-            newProperties.put(KEY_SERIALIZER_CLASS_CONFIG, keySerializer.getClass().getName());
-        if (valueSerializer != null)
-            newProperties.put(VALUE_SERIALIZER_CLASS_CONFIG, valueSerializer.getClass().getName());
-        return newProperties;
-    }
-
```

Comment:
We cannot remove this method since it's in a class that is part of Kafka's public API. We should probably deprecate it, but we need a KIP for that.

Reply:
@ijuma Should I file a issue for this PR if a KIP is required ?

Reply:
I think you could submit a KIP for the deprecation of the two methods in this class, but we can merge the other changes in the meantime.

Reply:
copy that!

Reply:
the follow-up https://issues.apache.org/jira/browse/KAFKA-10044